pkg/native/serializers/serializer_cdx.go

```diff
@@ -404,6 +404,17 @@ func (s *CDX) nodeToComponent(n *sbom.Node) *cdx.Component {
 		c.Copyright = n.GetCopyright()
 	}
 
+	if len(n.Properties) > 0 {
+		properties := []cdx.Property{}
+		for _, p := range n.Properties {
+			properties = append(properties, cdx.Property{
+				Name:  p.Name,
+				Value: p.Data,
+			})
+		}
+		c.Properties = &properties
+	}
```

Comment:
I don't fully understand the semantics of a pointer to a slice of values the upstream codebase likes to use, but would this work?

It should skip the for loop if `n.Properties` is empty, so I **_think_** the `if len(n.Properties) > 0` check could be removed. If it gets skipped, it just sets `c.Properties` to the address of the unallocated slice.

```suggestion
	var properties []cdx.Property
	for _, p := range n.Properties {
		properties = append(properties, cdx.Property{
			Name:  p.Name,
			Value: p.Data,
		})
	}

	c.Properties = &properties
```

Reply:
Here's a go playground [link](https://go.dev/play/p/Qzu8LOiJTFi). Hopefully it demonstrates properly what I'm asking about. 

Seems like a prevalent pattern in the serializer/unserializer files, maybe something like this could be applied to all instances of this pattern to improve cognitive complexity scores (and possibly performance)?

Reply:
Ah you are right, it's just an old habit I have. I've changed it and repushed.

That `sliceutil` package looks really useful, like something you should release on its own (wink wink)